browser/brave_stats/brave_stats_updater.cc

```diff
@@ -170,7 +178,7 @@ void BraveStatsUpdater::SetStatsThresholdCallback(
 
 GURL BraveStatsUpdater::BuildStatsEndpoint(const std::string& path) {
   auto stats_updater_url = GURL(usage_server_ + path);
-  DCHECK(stats_updater_url.is_valid());
```

Comment:
Could you instead just wrap this CHECK with `#if defined(OFFICIAL_BUILD)`?

Reply:
Also - while some anomaly might happen where this happens, does it need to be a proper `CHECK` (where it'll cause a fatal crash)? or would it be better to log the error (or something like this). Basically, if it got into this state, what would be the action to get it out of the state? (delete profile?)

Reply:
it would indicate we put out a release with a misconfigured stats server. this was requested by @bridiver, so he might have more to add.

Reply:
Just curious: would it be bad to just no-op if the server is an empty string, rather than doing this more complicated thing of checking whether it's a "development" build or not?

Reply:
@zenparsing probably not. that was what i initially did. i am implementing what @bridiver suggested in DM.

Reply:
Now I'm really curious! If the concern is shipping a build with a misconfigured `BRAVE_USAGE_SERVER`, then can we have a unittest that checks to make sure that `BRAVE_USAGE_SERVER` is valid if `defined(OFFICIAL_BUILD)`? It's a bit interesting that we need to add a helper to `brave_channel_info` to make this work.

Reply:
@bridiver hit me up and (DOH) - definitely my bad. If we build a production binary, we DEFINITELY want to crash if that is empty (it's a defective build). So I'd probably look at doing something like my original comment:
```
#if defined(OFFICIAL_BUILD)
CHECK(stats_updater_url.is_valid());
#endif
```

Reply:
I also think @zenparsing  has a good idea that catches the same thing without the CHECK. WDYT?

Reply:
pushed with the #ifdef guard

Reply:
With the `#ifdef` guard, I don't think the developer channel checks are needed anymore- are they?

Reply:
yeah seems fine. just tested and it's a no-op now. force pushed